### DISCUSSION TITLE
关于OpenAITranslator插件与ChatGPT在翻译质量上的差异

### DISCUSSION THREAD

**Discussion by Aryous** [nextai-translator/nextai-translator#657] [Q&A]
开发者你好！

我在使用这个插件的过程中发现其翻译质量相当高，非常感谢你开发出这么好用的插件。然而，我在使用ChatGPT进行相同翻译任务时，发现其翻译质量略有不同。比如下面的例子：

使用插件翻译【Every Cloud has a Silver Lining.】时，插件给出的结果是【乌云背后必有阳光】，这个翻译非常通顺且语义明确。但是，当我使用ChatGPT进行同样的翻译时，得到的结果为【每朵乌云背后都有一道银边】，尽管这个翻译在语义上是准确的，但它似乎缺少了一些文学性。

为了了解这种差异，我下载了您在GitHub上提供的源代码，并查看了系统提示（system prompt）：You are a translation engine that can only translate text and cannot interpret it. translate from English to Chinese: {text}。我尝试使用相同的prompt进行测试，但GPT仍然给出了【每朵乌云背后都有一道银边】的翻译结果。

我想请教您，是否存在其他原因导致了这种差异？如果是的话，我应该如何优化ChatGPT以获得更高质量的翻译结果呢？
![Snipaste_2023-05-01_21-39-10](https://user-images.githubusercontent.com/59194462/235459498-f8dbc428-102d-44bc-844b-5f89eb08ea72.png)
![Snipaste_2023-05-01_21-40-14](https://user-images.githubusercontent.com/59194462/235459703-0fa18400-6149-4692-9151-8976415afc76.png)

**Comment by Elegant-Spider** [comment]
### 发现了同样的问题，请教开发者。
用python写了个简单的调用：
`def translate_text(text):
    response = openai.ChatCompletion.create(
        model="gpt-3.5-turbo-16k-0613",
        temperature=1,
        max_tokens=1024,
        messages=[
            {"role": "system", "content": "You are a professional translation engine."},
            {"role": "user", "content": f"Please translate the following content to Chinese: {text}"}
        ]
    )
`
效果不如openai-translator
![屏幕截图 2023-07-28 233405](https://github.com/openai-translator/openai-translator/assets/139470389/6c2fa02b-b001-4e11-9256-8291a486adf9)

这是python调用结果：
尽管我们对肺癌的风险、发展、免疫控制和治疗选择有了进一步的认识，但它仍然是导致癌症死亡的首要原因。烟草吸烟仍然是肺癌发展的主要风险因素。非烟草的风险因素包括环境和职业暴露、慢性肺部疾病、肺部感染和生活方式因素。由于烟草仍然是肺癌的主要风险因素，疾病预防的重点是避免吸烟和戒烟。其他预防措施包括选择健康的饮食和保持积极的生活方式。未来的工作应重点放在戒烟宣传活动上，并更好地了解非吸烟者的疾病发展和治疗策略。

**Reply by chengfengfengwang** [reply]
概率问题吧？同样的参数每次回答的也不一样

**Comment by maltoze** [comment]
用的模型不一样，免费版的 ChatGPT 用的模型是 `text-davinci-002-render-sha`

**Comment by lingyunGH** [comment]
应该是要调参吧，但是参数的代码没找到

**Comment by lingyunGH** [comment]
![image](https://github.com/openai-translator/openai-translator/assets/16631249/c29ccd3f-c6e6-4149-bfac-f9e87d823fd4)

**Comment by jason19659** [comment]
搜一下呗，提示词都是开源的
